org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/jdt/ls/core/internal/hover/JavaElementLabelComposer.java

```diff
@@ -0,0 +1,1335 @@
+/**
+ * ***************************************************************************** Copyright (c) 2016
+ * IBM Corporation and others. All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which accompanies this
+ * distribution, and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * <p>Contributors: IBM Corporation - initial API and implementation
+ * *****************************************************************************
+ */
+package org.eclipse.jdt.ls.core.internal.hover;
+
+import java.util.jar.Attributes.Name;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.jdt.core.BindingKey;
+import org.eclipse.jdt.core.Flags;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IClassFile;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.IInitializer;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.ILocalVariable;
+import org.eclipse.jdt.core.IMember;
+import org.eclipse.jdt.core.IMemberValuePair;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.ISourceRange;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.ITypeParameter;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.Signature;
+import org.eclipse.jdt.core.SourceRange;
+import org.eclipse.jdt.internal.corext.dom.ASTNodes;
+import org.eclipse.jdt.internal.corext.util.JavaModelUtil;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.osgi.util.NLS;
+
+/**
+ * Implementation of {@link JavaElementLabels}.
+ *
+ * @since 3.5
+ */
+public class JavaElementLabelComposer {
+

```

Comment:
Why you don't use this class https://github.com/svor/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/hover/JavaElementLabelComposer.java ? As i understand you can add `org.eclipse.jdt.ls.core.internal.hover` to the manifest into Import-Package block and use it.

Reply:
That package is not exported and in consequence, you get a NoClassDefFoundException at runtime.